ssl/ssl_local.h

```diff
@@ -845,6 +845,30 @@ typedef struct tls_group_info_st {
     char is_kem;             /* Mode for this Group: 0 is KEX, 1 is KEM */
 } TLS_GROUP_INFO;
 
+typedef struct tls_sigalg_info_st {
+    char *tlsname;           /* Algorithm Name as in TLS specs */
+    char *tlsname_long;      /* Long Algorithm Name */
+    char *realname;          /* Algorithm Name according to provider */
+    char *algorithm;         /* Algorithm name to fetch */
+    char *oid;               /* OID of algorithm */
+    char *hash_algorithm;    /* Name of hash algorithm if any */
```

Comment:
These names and comments are quite confusing... and I can understand why, 'cause `SIGALG_LOOKUP` is a bit lacking.  What you should really have is something like this:

- TLS name (this is the "name" derived from the IANA algorithm description, currently stored in the *name* field of `SIGALG_LOOKUP`)
- Sigalg name (this corresponds to the *sigandhash* field in `SIGALG_LOOKUP`)
- Sig name (this is the pure signature algorithm, and corresponds to the *sig* field in `SIGALG_LOOKUP`)
- Hash name (this is the pure hash algorithm, and corresponds to the *hash* field in `SIGALG_LOOKUP`)
- Key type (this is the accepted key type for public key, *and currently has nothing in `SIGALG_LOOKUP`*)

All of these may be *different* (and may therefore also have each a distinct OID).  The lack of key type in `SIGALG_LOOKUP` is a historical omission, but should really be there...  for example, any of the ECDSA sigalgs should have "ECDSA" as the sig name and "EC" as the key type.

In this regard, short or long name shouldn't matter (if you need to check against the OBJ database, the sanest is probably to use something like `OBJ_txt2nid()` and compare the NIDs).  What should matter is that any name that is meant to be used to look up an algorithm should be fetchable (there should not be a distinct field for fetching)

Reply:
Logic & naming changed as per suggestions below. See https://github.com/openssl/openssl/pull/19312/commits/b036b423821ac7cab8a1ee48612c656105c71055

Reply:
What changes are expected to consider this resolved?

Reply:
As far as I can tell, this is resolved at this point.